go/DEVELOPER.md

```diff
@@ -0,0 +1,211 @@
+export PATH="$PATH:$HOME/go/bin"
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+# Install protobuf compiler
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-x86_64.zip
+unzip protoc-3.20.3-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler is installed
+protoc --version
+```
+
+**Dependencies installation for CentOS**
+
+```bash
+sudo yum update -y
+sudo yum install -y git gcc pkgconfig openssl openssl-devel unzip wget tar
+# Install Go
+wget https://go.dev/dl/go1.22.0.linux-amd64.tar.gz
+sudo tar -C /usr/local -xzf go1.22.0.linux-amd64.tar.gz
+export PATH="$PATH:/usr/local/go/bin"
+export PATH="$PATH:$HOME/go/bin"
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+# Install protobuf compiler
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-x86_64.zip
+unzip protoc-3.20.3-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler is installed
+protoc --version
+```
+
+**Dependencies installation for MacOS**
+
+```bash
+brew update
+brew install go make git gcc pkgconfig protobuf@3 openssl
+export PATH="$PATH:$HOME/go/bin"
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+```
+
+#### Building and installation steps
+
+Before starting this step, make sure you've installed all software requirements.
+
+1. Clone the repository:
+    ```bash
+    VERSION=0.1.0 # You can modify this to other released version or set it to "main" to get the unstable branch
+    git clone --branch ${VERSION} https://github.com/aws/glide-for-redis.git
+    cd glide-for-redis
+    ```
+2. Initialize git submodule:
+    ```bash
+    git submodule update --init --recursive
+    ```
+3. Install build dependencies:
+    ```bash
+    cd go
+    make install-build-tools
+    ```
+4. Build the Go wrapper:
+    ```bash
+    make build
+    ```
+5. Run tests:
+    1. Ensure that you have installed redis-server and redis-cli on your host. You can find the Redis installation guide at the following link: [Redis Installation Guide](https://redis.io/docs/install/install-redis/install-redis-on-linux/).
+    2. Execute the following command from the go folder:
+        ```bash
+        go test -race ./...
+        ```
+6. Install Go development tools with:
+
+    ```bash
+    make install-dev-tools
+    ```
+
+### Test
+
+To run tests, use the following command:
+
+```bash
+go test -race ./...
+```
+
+For more detailed test output, add the `-v` flag:
+
+```bash
+go test -race ./... -v
```

Comment:
do we need to go into detail here on specific flags for go test?  This is general knowledge of the tool and users can look this up.  If you want to include this, I'd suggest adding a blurb in the leading comment to say -v for verbose mode and -run <test_name> to run a specific test.

Reply:
I added this because without -v you go does not tell you which tests were run and because the Python DEVELOPER.md had a separate example command for demonstrating how to run a specific test. I can remove if you think that would be better though